tools/OMIcheck/OMIcheck.ps1

```diff
@@ -1,127 +1,460 @@
+        }
+    }
+}
+
+# If OMSLinuxAgent is installed, nudge the Azure Guest Agent to pickup the latest bits.
+function UpdateOMS($VM)
+{
+    $exts = Get-AzVMExtension -VMName $VM.Name -ResourceGroupName $VM.ResourceGroupName
+    $omsExt = $exts | where { ($_.Publisher -eq $OMSpublisher -and $_.Name -eq $OMSextName) }
+    if ($omsExt -ne $null)
+    {
+        if ($omsExt.ProvisioningState -eq "Succeeded")
+        {
+            # Trigger a goal state change for GA.
+            Set-AzVMExtension -VMName $VM.Name -ResourceGroupName $VM.ResourceGroupName -Location $omsExt.Location -Publisher $omsExt.Publisher -Name $omsExt.Name -ExtensionType $omsExt.ExtensionType -TypeHandlerVersion $omsExt.TypeHandlerVersion
+            Write-Host -ForegroundColor Green `t`t $VM.Name ": Set OMSLinuxAgent extension goal state for update"
+        }
+        else
+        {
+            Write-Host -ForegroundColor Gray `t`t $VM.Name ": OMSLinuxAgent extension is not in succeeded state. Skipping."
+        }
+    }
+}
+
+# If LAD is installed in VM scale set, nudge the Azure Guest Agent to pickup the latest bits.
+function UpdateLADinVMSS($VMss)
+{
+    Write-Host -ForegroundColor Red `t`t $VMss.Name ": VM scale set has vulnerable OMI version " $pkgVer
+
+    $ladExt = $VMss.VirtualMachineProfile.ExtensionProfile.Extensions | where { $_.Publisher -eq $LADpublisher -and $_.Type -eq $LADextName }
+    if ($ladExt -ne $null)
+    {
+        # Trigger a goal state change for GA.
+        $ladExt.AutoUpgradeMinorVersion = $true
+        Update-AzVmss -VirtualMachineScaleSet $VMss -ResourceGroupName $VMss.ResourceGroupName -VMScaleSetName $VMss.Name
+        # If the upgrade policy on the VMSS scale set is manual, start the upgrade on each instance manually.
+        if ($VMss.UpgradePolicy.Mode -eq "Manual")
+        {
+            foreach ($VMinstance in $VMinstances)
             {
-                Write-Host "`tResourceGroup: " $VM.ResourceGroupName
-                $PreviousRG = $VM.ResourceGroupName
+                Update-AzVmssInstance -ResourceGroupName $VMss.ResourceGroupName -VMScaleSetName $VMss.Name -InstanceId $VMinstance.InstanceId
             }
+        }
+        Write-Host -ForegroundColor Green `t`t $VMss.Name ": Set LAD extension is updated"
+    }
+}
+
+
+# If OMSLinuxAgent is installed in VM scale set, nudge the Azure Guest Agent to pickup the latest bits.
+function UpdateOMSinVMSS($VMss)
+{
+    Write-Host -ForegroundColor Red `t`t $VMss.Name ": VM scale set has vulnerable OMI version " $pkgVer
+
+    $omsExt = $VMss.VirtualMachineProfile.ExtensionProfile.Extensions | where { $_.Publisher -eq $OMSpublisher -and $_.Type -eq $OMSextName }
+    if ($omsExt -ne $null)
+    {
+        # Trigger a goal state change for GA.
+        $omsExt.AutoUpgradeMinorVersion = $true
+        Update-AzVmss -VirtualMachineScaleSet $VMss -ResourceGroupName $VMss.ResourceGroupName -VMScaleSetName $VMss.Name
```

Comment:
if `AutoUpgradeMinorVersion` is already true, will `Update-AzVmss` still do anything?

Reply:
No, good catch. I added -ForceRerun param.